### PR TITLE
drop `cv2` as dev dependency

### LINKED ISSUE #2592: Drop `cv2` from dev dependencies
## 🚀 Feature
The cv2 package is just used in one place at tests. So, we can figure out how to test the case without it
https://github.com/kornia/kornia/blob/216aa9d2f10f9300bffe952b1583f03b7f833be5/test/io/test_io_image.py#L4

## Motivation
reduce the number of dependencies at kornia

### PR DESCRIPTION
**Pull request by johnnv1**
#### Changes

Fixes #2592 

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by ducha-aiki — APPROVED** [review]

File: test/io/test_io_image.py [hunk 1]
```diff
@@ -45,12 +80,11 @@ def test_smoke(self, tmp_path: Path) -> None:
         assert img_th.shape[1:] == (height, width)
         assert str(img_th.device) == "cpu"
 
-    def test_device(self, device, tmp_path: Path) -> None:
+    def test_device(self, device, png_image) -> None:
```
Comment (edgarriba) [inline test/io/test_io_image.py:83]:
```suggestion
    def test_device(self, device, png_image: Path) -> None:
```

File: test/io/test_io_image.py [hunk 2]
```diff
@@ -45,12 +80,11 @@ def test_smoke(self, tmp_path: Path) -> None:
...
         height, width = 4, 5
-        img_np: np.ndarray = create_random_img8(height, width, 3)
+        create_random_img8(height, width, 3)
```
Comment (edgarriba) [inline test/io/test_io_image.py:85]:
```suggestion
```
?

File: test/io/test_io_image.py [hunk 3]
```diff
@@ -28,6 +29,40 @@ def create_random_img8_torch(height: int, width: int, channels: int, device=None
...
+    filename = _download_image(url, str(filename))
+    return filename
+
+
+@pytest.fixture(scope='class')
+def jpg_image(tmp_path_factory):
+    url = "https://github.com/kornia/data/raw/main/crowd.jpg"
+    filename = tmp_path_factory.mktemp("data") / "image.jpg"
+    filename = _download_image(url, str(filename))
+    return filename
```
Comment (edgarriba) [inline test/io/test_io_image.py:58]: return as Path ?

File: test/io/test_io_image.py [hunk 4]
```diff
@@ -28,6 +29,40 @@ def create_random_img8_torch(height: int, width: int, channels: int, device=None
...
+    with open(filename, "wb") as outfile:
+        outfile.write(bytesio.getbuffer())
+
+    return filename
+
+
+@pytest.fixture(scope='class')
+def png_image(tmp_path_factory):
+    url = "https://github.com/kornia/data/raw/main/simba.png"
+    filename = tmp_path_factory.mktemp("data") / "image.png"
```
Comment (edgarriba) [inline test/io/test_io_image.py:48]: why do you need to create `data` ? is not tmp_path_factory a new path every time the class implementing the fixture ? actually maybe we need to scope at session level

Comment (johnnv1) [reply test/io/test_io_image.py:48]: from https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-factory-fixture
> The tmp_path_factory is a session-scoped fixture which can be used to create arbitrary temporary directories from any other fixture or test.

the `tmp_path` is a new temp directory for each test itself

Comment (johnnv1) [reply test/io/test_io_image.py:48]: The idea here is to just download the PNG and JPG image one time per class or session, so it doesn't get heavy/slow

File: test/io/test_io_image.py (hunk 3 above)
Comment (johnnv1) [reply test/io/test_io_image.py:58]: yep, this way we can load from disk within kornia api

File: test/io/test_io_image.py [hunk 5]
```diff
@@ -75,12 +108,11 @@ def test_device(self, device, tmp_path: Path) -> None:
             (3, ImageLoadType.RGB32, torch.float32, 3),
         ],
     )
-    def test_load_image(self, tmp_path: Path, ext, channels, load_type, expected_type, expected_channels):
+    def test_load_image(self, images_fn, ext, channels, load_type, expected_type, expected_channels):
         height, width = 4, 5
-        img_np: np.ndarray = create_random_img8(height, width, channels)
+        create_random_img8(height, width, channels)
```
Comment (johnnv1) [inline test/io/test_io_image.py:113]:
```suggestion
```

File: test/io/test_io_image.py (hunk 3 above)
Comment (edgarriba) [reply test/io/test_io_image.py:58]: now it returns a `str` not `Path`

File: test/io/test_io_image.py [hunk 6]
```diff
@@ -28,6 +29,40 @@ def create_random_img8_torch(height: int, width: int, channels: int, device=None
...
+    # Download
+    bytesio = io.BytesIO(requests.get(url, timeout=60).content)
+    # Save file
+    with open(filename, "wb") as outfile:
+        outfile.write(bytesio.getbuffer())
+
+    return filename
+
+
+@pytest.fixture(scope='class')
```
Comment (edgarriba) [inline test/io/test_io_image.py:45]:
```suggestion
@pytest.fixture(scope='serssion')
```

File: test/io/test_io_image.py [hunk 7]
```diff
@@ -28,6 +29,40 @@ def create_random_img8_torch(height: int, width: int, channels: int, device=None
...
+def png_image(tmp_path_factory):
+    url = "https://github.com/kornia/data/raw/main/simba.png"
+    filename = tmp_path_factory.mktemp("data") / "image.png"
+    filename = _download_image(url, str(filename))
+    return filename
+
+
+@pytest.fixture(scope='class')
```
Comment (edgarriba) [inline test/io/test_io_image.py:53]:
```suggestion
@pytest.fixture(scope='session')
```

**Review by edgarriba — APPROVED** [review]

File: test/io/test_io_image.py (hunk 3 above)
Comment (johnnv1) [reply test/io/test_io_image.py:58]: true, got it

**Pull request merged**